score.py

```diff
@@ -128,6 +134,95 @@ def run_inference(images_with_header: List[ImageWithHeader],
     return segmentation
 
 
+def extract_zipped_dicom_series(zip_file_path: Path, extraction_folder: Path) -> None:
+    """
+    Unzip a zip file. Assume that it is a single DICOM series, and extract all
+    DICOM files into the extraction folder.
+
+    :param zip_file_path: Path to zip file.
+    :param extraction_folder: Path to extraction folder.
+    """
+    with zipfile.ZipFile(zip_file_path, 'r') as zip_file:
+        for zipped_file in zip_file.infolist():
+            # try to extract only DICOM files, but these need not have
+            # an extension
+            if not zipped_file.is_dir() and (
+                    zipped_file.filename.endswith('.dcm') or '.' not in zipped_file.filename):
+                # discard the path, if any, to just get the filename and suffix
+                zipped_file.filename = os.path.basename(zipped_file.filename)
+                zip_file.extract(zipped_file, str(extraction_folder))
+
+
+def convert_zipped_dicom_to_nifti(test_image: Path, model_folder: Path) -> Tuple[Path, Path]:
+    """
+    Given a zip file, extract DICOM series and convert to Nifti format.
+
+    :param test_image: Path to a zip file.
+    :param model_folder: Scratch folder to extract files into.
+    :return: Pair of path to Nifti file and path to folder of reference series.
+    """
+    reference_series_folder = model_folder / "temp_extraction"
+    extract_zipped_dicom_series(test_image, reference_series_folder)
+    nifti_filename = model_folder / "temp_nifti.nii.gz"
+    load_dicom_series_and_save(reference_series_folder, nifti_filename)
+    return nifti_filename, reference_series_folder
+
+
+def convert_rgb_colour_to_hex(colour: TupleInt3) -> str:
+    """
+    Config colours are stored as TupleInt3's, but DICOM-RT convert expects
+    hexadecimal strings. This function converts them into the correct
+    format.
+
+    :param colour: RGB colour as a TupleInt3.
+    :return: Colour formatted as a hex string.
+    """
+    return '{0:02X}{1:02X}{2:02X}'.format(colour[0], colour[1], colour[2])
+
+
+def convert_nifti_to_zipped_dicom_rt(nifti_file: Path, reference_series: Path, model_folder: Path,
+                                     config: SegmentationModelBase) -> Path:
+    """
+    Given a Nifti file and a reference DICON series, convert Nifti file to DICOM-RT file.
+
```

Comment:
Please extend documentation. In particular, say that you're created a ZIP file with a dicom-rt in it.
why is model_folder called that, if it's just scratch?